api-gateway/src/main/java/com/petclinic/bffapigateway/domainclientlayer/AuthServiceClient.java

```diff
@@ -70,19 +70,17 @@ public AuthServiceClient(
                         )
                 .bodyToMono(UserDetails.class);
     }
-
-    public Flux<UserDetails> createUsers (){
-        return webClientBuilder.build().post()
-                .uri(authServiceUrl + "/users")
-                .accept(MediaType.APPLICATION_JSON)
-                .retrieve().bodyToFlux(UserDetails.class);
-    }
-
-    public Mono<UserDetails> updateUser (final int userId, final UserDetails model){
+    public Mono<UserDetails> updateUser (final long userId, final Register model) {
         return webClientBuilder.build().put()
-                .uri(authServiceUrl + userId + model)
+                .uri(authServiceUrl + "/users/{userId}", userId)
+                .body(just(model), Register.class)
                 .accept(MediaType.APPLICATION_JSON)
-                .retrieve().bodyToMono(UserDetails.class);
+                .retrieve()
+                .onStatus(HttpStatus::is4xxClientError,
+                        n -> rethrower.rethrow(n,
+                                x -> new GenericHttpException(x.get("message").toString(), BAD_REQUEST))
+                        )
+                .bodyToMono(UserDetails.class);
```

Comment:
I cannot seem to find any tests related to this.
Can you please link me to them?